server/repository/src/database_settings.rs

```diff
@@ -49,7 +51,30 @@ impl DatabaseSettings {
 #[cfg(all(not(feature = "postgres"), not(feature = "memory")))]
 impl DatabaseSettings {
     pub fn connection_string(&self) -> String {
-        self.database_name.clone()
+        if self.database_name.ends_with(".sqlite") {
+            // just use DB if name ends in .sqlite
+            self.database_name.clone()
+        } else {
+            // first check if database exists on disk. If it does, we will use db filename as is without appending .sqlite
+            let db_exists = self.check_db_exists();
+            match db_exists {
+                true => self.database_name.clone(),
+                false => format!("{}.sqlite", self.database_name.clone()),
+            }
+        }
+    }
+
+    pub fn check_db_exists(&self) -> bool {
```

Comment:
Bit misleading method name. It checks if the db with name `database_name` exist. But it might only exist with ".sqlite" extension. Maybe just inline this one-liner (even though comment is quite verbose) to avoid confusion.

To avoid "if nesting" you can write the main method it like:

```rust
pub fn connection_string(&self) -> String {
        if self.database_name.ends_with(".sqlite") {
            // just use DB if name ends in .sqlite
            return self.database_name.clone()
        } 
         
        // first check if database exists on disk. If it does, we will use db filename as is without appending .sqlite
        let exists = INLINE DB CHECK HERE
        match exists {
            true => self.database_name.clone(),
            false => format!("{}.sqlite", self.database_name.clone()),
        }
}
```